### DISCUSSION TITLE
Use native CA store with OpenSSL on Windows

### DISCUSSION THREAD

**Discussion by talregev** [curl/curl#13068] [Q&A] [answered]
I want to compile http3 (USE_OPENSSL_QUIC) with openssl 3.2.0 and nghttp3 libs on windows.
The http3 is not relevant for the question beside that I would not use CURL_USE_SCHANNEL because it enable multi ssl feature along the openssl feature, and it cannot compile with http3 feature.

Can we implement for tool curl for windows when only openssl feature is enable: CURL_USE_OPENSSL that it will use os certificates files automatic?
Short code from stack overflow and code example that seems do it:
https://stackoverflow.com/questions/26893521/ca-certificate-directory-on-windows
https://ftp.netbsd.org/pub/NetBSD/NetBSD-current/src/external/bsd/wpa/dist/src/crypto/tls_openssl.c

I guess that already discuss, and I don't know why tool curl don't do it.
If not, I would like to suggest that and check if it possible.

**Comment by bagder** [comment, marked as answer]
See [--ca-native](https://curl.se/docs/manpage.html#--ca-native)

**Comment by talregev** [comment]
Thank you!
I check this option and it work. 
Any reason not to enable it always by default? 
Specially there is a "turn off" flag for that effect.

**Reply by bagder** [reply]
1. no one has argued for such a change
2. it would change behavior of existing use cases in unexpected ways

**Comment by talregev** [comment]
Another idea:
Can we add a compilation flag, that if compilation flag is enable. let say:  CURL_USE_CA_NATIVE
will imply this option automatic.
What do you think?
Thank you for your fast respond!

**Comment by talregev** [comment]
> Another idea: Can we add a compilation flag, that if compilation flag is enable. let say: CURL_USE_CA_NATIVE will imply this option automatic. What do you think? Thank you for your fast respond!

@bagder I search on curl to understand how I can implement CURL_USE_CA_NATIVE.
Is it CURL_WINDOWS_SSPI is similar? what happen on linux or other OS that is not windows when this flag is on?
Thank you for your help.
